### PR TITLE
Add support for the QueryUTxO via the new IPC API

### PR DESCRIPTION
**Pull request by dcoutts**
There are still quite a few queries to cover. This covers just the
QueryUTxO one. This requires exporting a UTxO type.

For now the UTxO type has no interesting instances but the newtype
wrapper is exposed which just uses the other API types in a simple way.
```
newtype UTxO era = UTxO (Map TxIn (TxOut era))
```

### CODE REVIEW

**Review by intricate — APPROVED** [review]

File: cardano-api/cardano-api.cabal [hunk 1]
```diff
@@ -17,18 +17,28 @@ library
...
-                        Cardano.Api.Shelley
                         Cardano.Api.Shelley.Genesis
                         Cardano.Api.TxSubmit
                         Cardano.Api.Typed
 
+                        --TODO: these are only exported for now, until the old
+                        -- IPC parts are removed and these modules can be
+                        -- exported via the normal export modules above
+                        Cardano.Api.IPC
+                        Cardano.Api.Query
```
Comment (intricate) [inline cardano-api/cardano-api.cabal:40]: I think you also need to remove these from `other-modules`

**Review by dcoutts — COMMENTED** [review]
@Jimbo4350 Thanks for the rebasing! Looks good.

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Review by newhoggy — APPROVED** [review]

**Pull request merged**